src/Articulate/BlogMlImporter.cs

```diff
@@ -219,7 +220,19 @@ private BlogMLDocument GetDocument(string fileName)
             foreach (var post in posts)
             {
                 //check if one exists
-                var postNode = allPostNodes.FirstOrDefault(x => x.GetValue<string>("importId") == post.Id);
+
+
+                IContent postNode;
+
+                //Use post.id if it's there
+                if (!String.IsNullOrWhiteSpace(post.Id))
+                {
+                    postNode = allPostNodes.FirstOrDefault(x => x.GetValue<string>("importId") == post.Id);                    
+                }
+                else //Use the "slug" (post name) if post.id is not there
+                {
+                    postNode = allPostNodes.FirstOrDefault(x => x.GetValue<string>("umbracoUrlName") != null && x.GetValue<string>("umbracoUrlName").StartsWith(post.Name.Content));
```

Comment:
Yeah, might be a good idea just to be safe, even if, in theory the `umbracoUrlName` should be copied from the content slug